ch-03/Solution39.java

```diff
@@ -0,0 +1,30 @@
+import java.util.Scanner;
+
+public class Solution39 {
+    public static void main(String[] args) {
+        final Scanner input = new Scanner(System.in);
+
+        System.out.println("Enter the first 9 digits of an ISBN");
+        final String isbnNineDigits = input.nextLine();
+
+        if (isbnNineDigits.length() != 9) {
+            System.out.println("You didn't enter exactly 9 digits, try it again.");
+        } else {
+            int[] allNumbers;
+            allNumbers = new int[9];
+            int d10 = 0;
+
+            for (int d = 0, digitsOfTheNumber = 0; d <= 8 && digitsOfTheNumber <= 8; d++, digitsOfTheNumber++) {

```

Comment:
There's no need for two loop variables, since they go through the same range. You're fine with just `d`.